tests/system/test_unary.py

```diff
@@ -76,59 +84,79 @@ def test_unary_with_dict(echo):
 
 
 def test_unary_error(echo):
-    message = 'Bad things! Bad things!'
+    message = "Bad things! Bad things!"
+    http_message = (
+        "POST http://localhost:7469/v1beta1/echo:echo: Bad things! Bad things!"
+    )
     # Note: InvalidArgument is from gRPC, BadRequest from http (no MTLS), InternalServerError from http (MTLS)
     # TODO: Reduce number of different exception types here.
-    with pytest.raises((exceptions.InvalidArgument, exceptions.BadRequest, exceptions.InternalServerError)) as exc:
-        echo.echo({
-            'error': {
-                'code': code_pb2.Code.Value('INVALID_ARGUMENT'),
-                'message': message,
-            },
-        })
-        assert exc.value.code == 400
-        assert exc.value.message == message
+    with pytest.raises(
+        (
+            exceptions.InvalidArgument,
+            exceptions.BadRequest,
+            exceptions.InternalServerError,
+        )
+    ) as exc:
+        echo.echo(
+            {
+                "error": {
+                    "code": code_pb2.Code.Value("INVALID_ARGUMENT"),
+                    "message": message,
+                },
+            }
+        )
+    err_message = message if "grpc" in str(echo.transport) else http_message
+    assert exc.value.code == 400
+    assert exc.value.message == err_message
```

Comment:
The TODO I suggest refers to separate work about cleaning error mesaging.

```suggestion
    # TODO: Fix rest errors so that we can use `message` rather than `err_message`
    assert exc.value.message == err_message
```

Reply:
Hmm, I don't think this is an issue with the rest error messages. Rather, we've been formatting the rest errors intentionally [here](https://github.com/googleapis/python-api-core/blob/b5ddd6236560f70405c0ea8614d67b27230341c5/google/api_core/exceptions.py#L508).

Reply:
Right, that's what I mean by separate work that we'll need to tackle as part of a separate project. But we can defer the TODO for now so we don't confuse the current PR.